cmake/clang-format.cmake

```diff
@@ -0,0 +1,21 @@
+file(GLOB_RECURSE
+  ALL_CXX_SOURCE_FILES
+  *.[ch] *.[chi]pp *.[chi]xx *.cppm *.cc *.hh *.ii *.[CHI]
+)
+
+find_program(CLANG_FORMAT 
+  NAMES clang-format clang-format-9.0
```

Comment:
I think in case `clang-format` (no suffix) is found, we need to check the version.

Reply:
Oh, I can remove this, I think. The default version should also add the numbered version. Lemme see

Reply:
This is indeed the case on Ubuntu, but I can not say for sure about other OSes. Wasn't Alpine using suffix-less binary?

Reply:
On the docker images, clang-format installed both the numbered target as well as the generic sans-suffix. I don't know if that's the case with clang-format from source. I think we can leave that as a corner case for the advanced user to fix. What do you think?

Reply:
>  I think we can leave that as a corner case for the advanced user to fix

I am fine with such approach, however I'd at least drop some hints to help that advanced user. Like maybe add a comment that explains why sans-suffix `clang-format` is not in the search list and what needs to be done if someone wants to add it.

Reply:
> Wasn't Alpine using suffix-less binary?
Yes it is. So I need to handle this. Let's see. I think I'll attack the doc exclusion first

Reply:
You can copy the version parsing code from [here](https://github.com/BlueBrain/git-cmake-format/blob/master/FindClangFormat.cmake).

Reply:
Thanks for the link. So this regex `^clang-format version .*` didn't match, I decided not to use that logic and found a shorter method (since I didn't need minor and patch numbers)